client/state/media/reducer.js

```diff
@@ -144,43 +145,28 @@ export const errors = ( state = {}, action ) => {
 	return state;
 };
 
-function applyToManager( state, siteId, method, createDefault, ...args ) {
-	if ( ! state[ siteId ] ) {
-		if ( ! createDefault ) {
-			return state;
-		}
-
-		return {
-			...state,
-			[ siteId ]: new MediaQueryManager()[ method ]( ...args ),
-		};
-	}
-
-	const nextManager = state[ siteId ][ method ]( ...args );
-
-	if ( nextManager === state[ siteId ] ) {
-		return state;
-	}
-
-	return {
-		...state,
-		[ siteId ]: nextManager,
-	};
-}
-
 export const queries = ( state = {}, action ) => {
 	switch ( action.type ) {
 		case MEDIA_RECEIVE: {
 			const { siteId, media, found, query } = action;
-			return applyToManager( state, siteId, 'receive', true, media, { found, query } );
+			return withQueryManager(
+				state,
+				siteId,
+				() => new MediaQueryManager(),
+				( manager ) => manager.receive( media, { found, query } )
+			);
 		}
 		case MEDIA_DELETE: {
 			const { siteId, mediaIds } = action;
-			return applyToManager( state, siteId, 'removeItems', true, mediaIds );
+			return withQueryManager( state, siteId, false, ( manager ) =>
+				manager.removeItems( mediaIds )
+			);
 		}
 		case MEDIA_ITEM_EDIT: {
 			const { siteId, mediaItem } = action;
-			return applyToManager( state, siteId, 'receive', true, mediaItem, { patch: true } );
+			return withQueryManager( state, siteId, false, ( manager ) =>
+				manager.receive( mediaItem, { patch: true } )
+			);
```

Comment:
How did you decide that these should not create a new query manager?

Reply:
My reasoning was that the `MEDIA_ITEM_EDIT` and `MEDIA_DELETE` actions act on an existing list of media items. So if the list is not there in the first place, they have no work to do.

@flootr Can you confirm that's true especially for `MEDIA_ITEM_EDIT`? The `editMedia` thunk is rather complicated, but still it seems certain to me that it's ever called only when there is a valid list item that `getMediaItem` would return, and wouldn't really work correctly otherwise.